quiche/src/lib.rs

```diff
@@ -7543,6 +7543,20 @@ impl Connection {
     fn get_send_path_id(
         &self, from: Option<SocketAddr>, to: Option<SocketAddr>,
     ) -> Result<usize> {
+        // When a PTO timer expires, a sender MUST send at least one ack-eliciting packet.
+        let mut pto = self
+            .paths
+            .iter()
+            .filter(|(_, p)| from.is_none() || Some(p.local_addr()) == from)
+            .filter(|(_, p)| to.is_none() || Some(p.peer_addr()) == to)
+            .filter(|(_, p)| p.usable())
+            .filter(|(_, p)| p.recovery.loss_probes.iter().any(|&x| x > 0))
+            .map(|(pid, _)| pid);
+
+        if let Some(pid) = pto.next() {
+            return Ok(pid);
+        }
+
```

Comment:
I agree that this code should probably be part of `get_send_path_id()`, but I wonder if it would make sense to consider that heuristic *after* sending data (i.e., with the code forcing the sending of ACK_MP frames on all paths).